packages/dashboard-backend/src/devworkspaceClient/services/gitConfigApi/__tests__/index.spec.ts

```diff
@@ -29,198 +29,266 @@ const responseBody = {
+      });
 
-    it('should throw when failed to patch the ConfigMap', async () => {
-      spyPatchNamespacedConfigMap.mockRejectedValueOnce('some error');
+      it('should throw when failed to patch the ConfigMap', async () => {
+        spyPatchNamespacedConfigMap.mockRejectedValueOnce('some error');
 
-      const newGitConfig = {
-        gitconfig: {
-          user: {
-            email: 'user-2@che',
-            name: 'User Two',
+        const newGitConfig = {
+          gitconfig: {
+            user: {
+              email: 'user-2@che',
+              name: 'User Two',
+            },
           },
-        },
-      } as api.IGitConfig;
+        } as api.IGitConfig;
 
-      try {
-        await gitConfigApiService.patch(namespace, newGitConfig);
+        try {
+          await gitConfigApiService.patch(namespace, newGitConfig);
 
-        // should not reach here
-        expect(false).toBeTruthy();
-      } catch (e) {
-        expect((e as Error).message).toBe(
-          'Unable to update gitconfig in the namespace "user-che": some error',
-        );
-      }
+          // should not reach here
+          expect(false).toBeTruthy();
+        } catch (e) {
+          expect((e as Error).message).toBe(
+            'Unable to update gitconfig in the namespace "user-che": some error',
+          );
+        }
 
-      expect(spyReadNamespacedConfigMap).toHaveBeenCalledTimes(1);
-      expect(spyPatchNamespacedConfigMap).toHaveBeenCalledTimes(1);
-    });
+        expect(spyReadNamespacedConfigMap).toHaveBeenCalledTimes(1);
+        expect(spyPatchNamespacedConfigMap).toHaveBeenCalledTimes(1);
+      });
 
-    it('should throw when conflict detected', async () => {
-      spyReadNamespacedConfigMap.mockResolvedValueOnce({
-        body: {
-          metadata: {
-            resourceVersion: '2',
-          },
-          data: {
-            gitconfig: `[user]
+      it('should throw when conflict detected', async () => {
+        spyReadNamespacedConfigMap.mockResolvedValueOnce({
+          body: {
+            metadata: {
+              resourceVersion: '2',
+            },
+            data: {
+              gitconfig: `[user]
 name="User Two"
 email="user-2@che"
 `,
+            },
+          } as V1ConfigMap,
+          response: {} as IncomingMessage,
+        });
+
+        const newGitConfig = {
+          gitconfig: {
+            user: {
+              email: 'user-2@che',
+              name: 'User Two',
+            },
           },
-        } as V1ConfigMap,
-        response: {} as IncomingMessage,
+          resourceVersion: '1',
+        } as api.IGitConfig;
+
+        try {
+          await gitConfigApiService.patch(namespace, newGitConfig);
+
+          // should not reach here
+          expect(false).toBeTruthy();
+        } catch (e) {
+          expect((e as Error).message).toBe(
+            'Conflict detected. The gitconfig was modified in the namespace "user-che".',
+          );
+        }
```

Comment:
```suggestion
      let errorMessage = '';
      try {
        await gitConfigApiService.patch(namespace, newGitConfig);
      } catch (e: unknown) {
        errorMessage = (e as Error).message;
      }
      expect(errorMessage).toBe(
        'Conflict detected. The gitconfig was modified in the namespace "user-che".',
      );
```

Reply:
I suggest to refactor this code. It will help to understand expected behavior after each executed test step.

WDYT?